### DISCUSSION TITLE
Unable ejecting safely an external USB HD

### DISCUSSION THREAD

**Discussion by KerimF1** [WinMerge/winmerge#734] [General]
First, thanks to all who contributed in developing the professional easy-to-use program, ‘WinMerger’ .

I noticed that after closing ‘WinMerger’ I can’t eject (disconnect) safely the USB HD (used for backup). A warning message pops up saying that it is still in use by a program. This happens even if ‘WinMerger’ (version 2.16.10.0 x86) was the only program I run and closed after booting the laptop. So to free and eject safely the external HD in this case, I have to restart the laptop and reboot Windows 7.

Any idea?

Thank you.
Kerim

**Comment by sdottaka** [comment]
Even if you exit WinMerge, the process may not disappear completely. Can you see if WinMergeU.exe is still alive with taskmgr.exe when an external USB HD cannot be ejected?

I'm aware of the problem that WinMerge sometimes doesn't quit completely, but it doesn't always reproduce, so I'd like to know the reproduction conditions.

**Reply by KerimF1** [reply]
Thank you for replying,

You are right. It seems that WinMerge 'sometimes' (not always) doesn't quit completely.
I am saying this because after I started this topic, I needed 'WinMerge' to compare many folders/files on the main HD to the ones on the external backup HD.  This time and after closing the program, the USB HD was free to be ejected safely.

As you said, it is necessary to discover first the conditions which reproduce both cases.

My first impression, so I may be wrong, is that all result windows (mainly the ones that are related to the USB HD) may need to be closed first before closing the program.

**Reply by toni1elias** [reply]
Hello.
I have the same problem, but I couldn't recreate it, because it does not happen all the time.

**Comment by sdottaka** [comment]
The problem with WinMerge sometimes not quitting has been fixed in version 2.16.15.

**Reply by KerimF1** [reply]
Thanks for the good news.
By the way, what could be the best way to replace the older version (2.16.10) with the new one (2.16.15)?
I run W7 32.
Thank you.

**Reply by toni1elias** [reply]
I have version 2.16.14 and the USB problem is not fixed in it. Will test the 2.16.15 version when it's stable.

**Comment by sdottaka** [comment]
Overwrite the installation with WinMerge-2.16.15-Setup.exe.

You may be asked to reboot your OS to update the shell extension.

This is a strange way to do it, but if you reboot your OS before installing and install WinMerge without opening the Explorer context menu, you should not be asked to reboot your OS.

**Reply by KerimF1** [reply]
Thank you.
I guess WinMerge-2.16.15-Setup.exe is for Windows x32 while WinMerge-2.16.15-x64-Setup.exe is for x64.

**Reply by Gitoffthelawn** [reply]
Or you can use the portable version and forego the installer.

If you choose this route:

1. Backup your WinMerge settings.
2. Uninstall WinMerge.
3. Copy the portable version into a folder and create a shortcut for it.
4. Restore your WinMerge settings if they get removed by Step 2.

**Reply by KerimF1** [reply]
Thank you.

**Reply by Gitoffthelawn** [reply]
@KerimF1 You're welcome.
